src/coreclr/jit/valuenum.cpp

```diff
@@ -8006,10 +8006,11 @@ void Compiler::fgValueNumberTree(GenTree* tree)
                                         rhsVNPair.SetBoth(vnStore->VNForExpr(compCurBB, lclVarTree->TypeGet()));
                                     }
 
+                                    lclDefSsaNum = GetSsaNumForLocalVarDef(lclVarTree);
```

Comment:
For a tree like:
```
N005 ( 11, 10) [001400] -A------R----             *  ASG       long
N004 (  6,  5) [001401] *------N-----             +--*  IND       long
N003 (  3,  3) [001402] -------------             |  \--*  ADDR      byref  Zero Fseq[_00]
N002 (  3,  2) [001403] U------N-----             |     \--*  LCL_VAR   struct<System.Runtime.Intrinsics.Vector128`1[Byte], 16> V45 tmp38        ud:3->4
N001 (  1,  1) [001404] -------------             \--*  LCL_VAR   long   V69 tmp62        u:2 (last use)
```
SSA was working correctly and printing that `001403` is using SSA-3 and defining SSA-4. However, this code, for some reason, was writing the result as a define for SSA-3.

So when at the next tree we were asking for `GetPerSsaData(4)` we were not getting an initialized value.

Reply:
Maybe this comes up now because we no longer set `lvOverlappingFields` in the importer?

Reply:
It looks like this could be moved to come before line 8051, as that is the first use of this value,